spark3/src/main/java/org/apache/iceberg/spark/source/SparkBatchScan.java

```diff
@@ -208,6 +209,38 @@ public Statistics estimateStatistics() {
     return new Stats(sizeInBytes, numRows);
   }
 
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    SparkBatchScan that = (SparkBatchScan) o;
+    // compare Spark schemas to ignore field ids
+    StructType thisSchema = SparkSchemaUtil.convert(expectedSchema);
+    StructType thatSchema = SparkSchemaUtil.convert(that.expectedSchema);
+    return caseSensitive == that.caseSensitive &&
```

Comment:
if they aren't case sensitive aren't they equal if the lowercased names are equivelent?

Ie 
(insensitive) == (insenstive) && (TaBle).equals(table)?

If we just want to check the fields being equal I understand that too

Reply:
My initial thought was that filter expression binding depends on whether this is case sensitive or not. But since we get this value from Spark and don't expect it to change (let alone in a single query), it seems unnecessary. I'm also not able to think of a case where a difference would matter without causing an exception.

I'll remove this.